### PR TITLE
[Feature] Update fakeCAS for [OSF#6438]

### PR DESCRIPTION
**Pull request by cslzchen**
### Purpose

With our new auth login for OSF and CAS, fakeCAS is updated to mimic the behavior of CAS.
### Changes

Simple but elegant pages including:
- Login (Only username required)
- User not exist
- User not registered
- Invalid verification request
- Unauthorized services (Disabled)
### Side Effects

No
### Ticket

https://openscience.atlassian.net/browse/OSF-6438
### Related PR

https://github.com/CenterForOpenScience/cas-overlay/pull/18
https://github.com/CenterForOpenScience/osf.io/pull/5804

### CODE REVIEW

File: static.go [hunk 1]
```diff
@@ -1,49 +1,64 @@
...
-                <a href="https://github.com/CenterForOpenScience/centerforopenscience.org/blob/master/TERMS_OF_USE.md">Terms of Use</a> |
-                <a href="https://github.com/CenterForOpenScience/centerforopenscience.org/blob/master/PRIVACY_POLICY.md">Privacy Policy</a>
-            </p>
-        </div>
+    {{if .NotValid}}
+      <p>Invalid request. User does not exist or invalid/expired validation key.</p>
+    {{end}}
+    {{if .LoginForm}}
+    <div id="forms">
+      <form id="login" action="/login?service=http://localhost:5000" method="post">
```
Comment (chrisseto) [inline static.go]: `service` will need to be URL encoded

File: static.go [hunk 2]
```diff
@@ -1,49 +1,64 @@
...
+          <input id="persistence" type="checkbox" value="true" checked/>
+          <label id="for-persistence">Stay Signed In</label>
+        </section>
+      </form>
+    </div>
+    {{end}}
+    </br>
+    <div id="links">
+      <section>
+        <a id="forgot-password" href="http://localhost:5000/forgotpassword/">Forgot Your Password?</a></br>
```
Comment (chrisseto) [inline static.go]: `http://localhost:5000/` should be tempalted

File: static/login.html [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          <input id="persistence" type="checkbox" value="true" checked/>
+          <label id="for-persistence">Stay Signed In</label>
+        </section>
+      </form>
+    </div>
+    {{end}}
+    </br>
+    <div id="links">
+      <section>
+        <a id="forgot-password" href="http://localhost:5000/forgotpassword/">Forgot Your Password?</a></br>
```
Comment (chrisseto) [inline static/login.html]: Template this URL

File: static/login.html [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        <a id="institution-login" href="http://localhost:5000/login/?campaign=institution&redirect_url=http://localhost:5000/">Login Through Your Institution</a></br>
```
Comment (chrisseto) [inline static/login.html]: `redirect_url` will need to be URL encoded

File: views.go [hunk 5]
```diff
@@ -9,31 +9,107 @@ import (
 	"strings"
 )
 
-func Login(c echo.Context) error {
-	redir, err := url.Parse(c.FormValue("service"))
+func ValidateService(c echo.Context) *url.URL {
+	service, err := url.Parse(c.QueryParam("service"))
```
Comment (chrisseto) [inline views.go]: You can just return service here. If `err != nil` then `service == nil`

File: views.go [hunk 6]
```diff
@@ -9,31 +9,107 @@ import (
...
+		// }
+	}
+	return nil
+}
+
+func LoginPOST(c echo.Context) error {
+	data := new(TemplateData)
+
+	service := ValidateService(c)
+	if service == nil {
```
Comment (chrisseto) [inline views.go:16]: Stylistic thing, https://golang.org/doc/effective_go.html#if

``` go
if service := ValidateService(c); service == nil {
...
```

File: views.go [hunk 7]
```diff
@@ -9,31 +9,107 @@ import (
...
+		data.LoginForm = true
+		return c.Render(http.StatusOK, "login", data)
+	}
 
 	result := User{}
-
-	if err = UserCollection.Find(bson.M{"username": c.FormValue("username")}).One(&result); err != nil {
-		fmt.Println("User", c.FormValue("ticket"), "not found.")
-		return c.NoContent(http.StatusNotFound)
+	err = UserCollection.Find(bson.M{"username": c.FormValue("username")}).One(&result)
```
Comment (chrisseto) [inline views.go]: Same stylistic nit pik. Assign `err` in the `if`

File: views.go (hunk 6 above)
Comment (cslzchen) [inline views.go:16]: `service` is used outside this `if`, will throw `undefined` error in this style, same for `username` and `verification_key`.

File: static.go (hunk 1 above)
Comment (cslzchen) [inline static.go]: do I need to encode if I use `url.Parse("http://" + *Host + "/login?service=" + service)`?

**Pull request merged**